src/main/kotlin/de/sldk/mc/PrometheusExporter.kt

```diff
@@ -14,14 +18,18 @@ class PrometheusExporter : JavaPlugin() {
     override fun onEnable() {
         config.loadDefaultsAndSave()
         config.enableConfiguredMetrics()
-        startMetricsServer()
+
+        val healthChecks = ConcurrentHealthChecks.create()
+        getServer().servicesManager.register(HealthChecks::class.java, healthChecks, this, ServicePriority.Normal)
```

Comment:
@JarvisCraft I know that it's been a while since you've written this... This PR contains your health check implementation for the Prometheus exporter. I rebased it and just noticed that I have no idea how this actually works.

How would a custom plugin add a health check to this "service"? Don't we need to provide the `HealthChecks` and `HealthCheck` classes as a library for this to work in third party plugins?

Reply:
<img src="https://github.com/sladkoff/minecraft-prometheus-exporter/assets/7693005/a904bd4f-14d1-433a-bac6-55597eee0458" width="100" />

Any plugin may access this `HealthCheck` instance by depending on this plugin and calling [`ServicesManager#load(String)`][1].

For example, I may want to do the following in my plugin:

```kotlin
server.servicesManager.load(HealthChecks::class.java)!!
    .add(myCustomHealthChecl)
```

> Don't we need to provide the HealthChecks and HealthCheck classes as a library for this to work in third party plugins?

Yup, this would be the best to split the API and implementation for this purpose (with the API being available via Maven repository), although generally it is not a strict requirement since the plugin JAR can be used as a compile-time dependency to link to this class.

[1]: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/plugin/ServicesManager.html#load(java.lang.Class)

Reply:
BTW, it seems that the line can be simplified:

```suggestion
        server.servicesManager.register(HealthChecks::class.java, healthChecks, this, ServicePriority.Normal)
```

Reply:
Nice, thanks for the clarification. I added some instructions about this to the README (feel free to add to it in a separate PR). A maven distribution would be nice in the future but I guess this will do for now. 

Thanks for your work on this and sorry again for the delay.

Reply:
You are welcome :heart: